pkg/tsdb/azuremonitor/types/types.go

```diff
@@ -139,9 +140,10 @@ type AzureMonitorJSONQuery struct {
 // AzureMonitorDimensionFilter is the model for the frontend sent for azureMonitor metric
 // queries like "BlobType", "eq", "*"
 type AzureMonitorDimensionFilter struct {
-	Dimension string `json:"dimension"`
-	Operator  string `json:"operator"`
-	Filter    string `json:"filter"`
+	Dimension string   `json:"dimension"`
+	Operator  string   `json:"operator"`
+	Filter    string   `json:"filter"`
+	Filters   []string `json:"filters,omitempty"`
```

Comment:
it doesn't sound right to me that we have Filter and Filters. Is that necessary? Why is not a Filter a slice of one element? It sounds like you are doing that to support the current queries but if that's the case we would need to:
 - Add a comment saying that using `filter` is deprecated, and migrate here to use `filters` (wherever this struct is being populated).
 - In the QueryEditor, migrate it too so after editing a query, it is now using the list.

Does that make sense? That way we can remove the special handling when Filter is set.

Reply:
A migration would make the most sense here to support current queries and also implement the new functionality. I'll look into this. This would simplify some things as the `Filter` property is essentially only used now for the `sw` operator.

Reply:
cool, let me know if you find any issue with that.